backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/monitoring/VirtMonitoringStrategy.java

```diff
@@ -45,19 +47,28 @@ public class VirtMonitoringStrategy implements MonitoringStrategy {
     @Inject
     private Instance<IVdsEventListener> eventListener;
 
+    @Inject
+    private ImageTransferDao imageTransferDao;
+
     @Inject
     public VirtMonitoringStrategy(ClusterDao clusterDao,
             VdsDao vdsDao,
             VmDao vmDao,
-            VmDynamicDao vmDynamicDao) {
+            VmDynamicDao vmDynamicDao,
+            ImageTransferDao imageTransferDao) {
         this.clusterDao = clusterDao;
         this.vdsDao = vdsDao;
         this.vmDao = vmDao;
         this.vmDynamicDao = vmDynamicDao;
+        this.imageTransferDao = imageTransferDao;
     }
 
     @Override
     public boolean canMoveToMaintenance(VDS vds) {
+        List<ImageTransfer> transfers = imageTransferDao.getByVdsId(vds.getId());
+        if (!transfers.stream().allMatch(ImageTransfer::isPausedOrFinished)) {
+            return false;
```

Comment:
how about adding a log so we'll know that's the reason for blocking the transition to maintenance?

Reply:
but that's not the reason here, it's because there's an ongoing image transfer, right? I wouldn't put it in 'debug' mode, it can be in 'info' mode - it's not that common to happen and when it does, we may want to know about it without changing the log level

Reply:
I Changed the log message. @ahadas please review again